feature(admin): add comprehensive user management APIs

Cuonq2912 opened the pull request:
- Add admin endpoints for user CRUD operations (GET, POST, PUT, DELETE)
- Implement account locking/unlocking functionality
- Add pagination support and DTO mappings

- Another change was implemented:

+ Removed unused bean [userService](https://github.com/KhuongSpring/COREVO/blob/feature/implement-admin-user-apis/corevo/src/main/java/com/example/corevo/service/impl/JwtServiceImpl.java) from JwtService to avoid creating a circular dependency between JwtAuthenticationFilter → JwtServiceImpl → UserServiceImpl → SecurityConfig → JwtAuthenticationFilter
+ Added isLocked field in the [User](https://github.com/KhuongSpring/COREVO/blob/feature/implement-admin-user-apis/corevo/src/main/java/com/example/corevo/domain/entity/User.java) entity to implement soft delete

KhuongSpring reviewed (commented):
1. In UserController (Admin method):
- Because you already have permit ROLE_ADMIN in SecurityConfig, and ADMIN_END_POINT, so you don't need to use @PreAuthorize in Controller. But it's okay if you don't want to change code.
2. In CreateUserRequestDto
- If you have @Email, you don't need to use @NotBlank.
3. In UserService
- In create user, you need to check phone existed
4. In UrlConstant
- Why is endpoint "/users/{userId}" repeats too much? You can reuse it. 
- One advice is you should write exactly enpoint. Example:
    
        private static final String PRE_FIX = "/admin";
        public static final String GET_USERS = PRE_FIX + "/users";
        public static final String GET_USER = PRE_FIX + "/user/{userId}";
        public static final String CREATE_USER = PRE_FIX + "/create-user";
        public static final String UPDATE_USER = PRE_FIX + "/update-user/{userId}";
        public static final String DELETE_USER = PRE_FIX + "/delete-user/{userId}";

        public static final String LOCK_USER = PRE_FIX + "/lock-user/{userId}";
        public static final String UNLOCK_USER = PRE_FIX + "/unlock-user/{userId}";
5. You need to init one account admin for testing API...

Cuonq2912 commented:
@KhuongSpring. I have updated code with all your suggestions: 
- Remove @PreAuthorize in UserController
- Added phone existence check in userServiceImpl
- Refactored UrlConstant to reuse for endpoints.
- Implemented admin account initialization on application startup
- Replaced all ModelMapper with MapStruct mappers
After researching, I found that @Email only validates email format but doesn't check for null/empty values. Therefore, I kept @NotBlank alongside @Email to ensure the email field is both present and properly formatted.
Thanks.
